packages/react-ui-components/src/stories/Components/LabelsDropdown.stories.tsx

```diff
@@ -0,0 +1,163 @@
+import React, { useCallback, useMemo, useState } from 'react';
+import { ComponentStory } from '@storybook/react';
+import { KvLabelsDropdown, KvSelectOption } from '../../components';
+import { ISelectMultiOptions } from '@kelvininc/ui-components/dist/types/components/select-multi-options/select-multi-options.types';
+import { get, isEmpty } from 'lodash';
+import { searchMultiLevelDropdownOptions } from './helpers/dropdown.helper';
+
+KvLabelsDropdown.displayName = 'KvLabelsDropdown';
+
+const getAllChildrenValues = (options: ISelectMultiOptions): string[] => {
+	if (!options || Object.values(options).length <= 0) {
+		return [];
+	}
+	return Object.values(options).reduce<string[]>((acc, option) => {
+		if (isEmpty(option.children)) {
+			acc.push(option.value);
+		} else {
+			acc = acc.concat(getAllChildrenValues(option.children));
+		}
+		return acc;
+	}, []);
+};
+
+export default {
+	title: 'Inputs/Dropdown/Labels',
+	component: 'kv-labels-dropdown',
+	argTypes: {
+		isOpen: { control: { type: 'boolean' } },
+		options: { control: { type: 'object' } },
+		selectedOptions: { control: { type: 'object' } },
+		filteredOptions: { control: { type: 'object' } },
+		noDataAvailableLabel: { control: { type: 'text' } },
+		searchable: { control: { type: 'boolean' } },
+		searchValue: { control: { type: 'text' } },
+		minHeight: { control: { type: 'text' } },
+		maxHeight: { control: { type: 'text' } }
+	},
+	parameters: {
+		notes: require('@ui-notes/labels-dropdown/readme.md')
+	}
+};
+
+const LabelsDropdownTemplate: ComponentStory<typeof KvLabelsDropdown> = args => {
+	const [options, setOptions] = useState(args.options || {});
+	const [selectedOptions, setSelectedOptions] = useState(args.selectedOptions);
+	const [searchTerm, setSearchTerm] = useState<string | null>(null);
+
+	const filteredOptions = useMemo(() => searchMultiLevelDropdownOptions(searchTerm ?? '', options ?? {}), [searchTerm, options]);
+
+	const onSelectAll = useCallback(() => {
+		const newSelectedOptions = getAllChildrenValues(options ?? {}).reduce<Record<string, boolean>>((acc, cur) => {
+			acc[cur] = true;
+			return acc;
+		}, {});
+
+		setSelectedOptions(newSelectedOptions);
+	}, [options, getAllChildrenValues]);
+
+	const onClearAll = useCallback(() => setSelectedOptions({}), []);
+
+	const onSearchChange = useCallback(({ detail: searchedLabel }: CustomEvent<string>) => {
+		setSearchTerm(searchedLabel);
+	}, []);
+
+	const onOptionsSelected = useCallback(({ detail }: CustomEvent<Record<string, boolean>>) => {
+		setSelectedOptions(detail);
+	}, []);
+
+	const onAddNewLabel = ({ detail }: CustomEvent<string>) => {
+		const existOption = get(options, ['recents', detail]);
+		if (!existOption) {
+			setOptions(prev => ({
```

Comment:
using the `merge` function I can't re-order the object keys as intended in a one liner, I can copy from `prev` but this would defeat the purpose of using merge no?

Reply:
I didn't see that you were reordering the object keys. Nvmd 👍